### PR TITLE
dynamic civ names

### PR DESCRIPTION
**Pull request by MattOttawa**

### CODE REVIEW

File: Assets/Python/Revolution/Gameready/DynamicCivNames.py [hunk 1]
```diff
@@ -0,0 +1,763 @@
...
+    
+    eLeader = pPlayer.getLeaderType()
+    
+    bFemaleLeader = self.isFemaleLeader(eLeader)
+
+    iLanguage = gc.getGame().getCurrentLanguage()
+    bFrench = iLanguage == 1 #0 - English, 1 - French, 2 - German, 3 - Italian, 4 - Spanish
+
+    eGovCivic = pPlayer.getCivics(gc.getInfoTypeForString("CIVICOPTION_GOVERNMENT"))
+    ePowerCivic = pPlayer.getCivics(gc.getInfoTypeForString("CIVICOPTION_POWER"))
```
Comment (MattOttawa) [inline Assets/Python/Revolution/Gameready/DynamicCivNames.py:413]: Here are the civic categories the names are based on. What should it be now?

File: Assets/Python/Revolution/Gameready/DynamicCivNames.py [hunk 2]
```diff
@@ -0,0 +1,763 @@
...
+          if( self.LOG_DEBUG and bVerbose ) : CvUtil.pyPrint("Names - keeping prior name")
+          newName = curDesc
+        elif( 50 > game.getSorenRandNum(100,'Rev: Naming') ) :
+          newName = localText.getText("TXT_KEY_MOD_DCN_SOC_REP", ())%(curShort)
+        else :
+          newName = localText.getText("TXT_KEY_MOD_DCN_PEOPLES_REP", ())%(curShort)
+      elif( RevUtils.getDemocracyLevel(iPlayer)[0] == -8 ) :
+        if( localText.getText("TXT_KEY_MOD_DCN_RUSSIAN_MATCH", ()) in curAdj ) :
+          curAdj = localText.getText("TXT_KEY_MOD_DCN_SOVIET", ())
+        newName = localText.getText("TXT_KEY_MOD_DCN_UNION", ())%(curAdj)
```
Comment (MattOttawa) [inline Assets/Python/Revolution/Gameready/DynamicCivNames.py:526]: Loads of xml text keys that need to be checked...

**Comment by raxo2222** [comment]
This could be untied to Revolutions, that is always active.
All GOVERMENT civics:

ANARCHISM
CHIEFDOM
DESPOTISM
MONARCHY
REPUBLIC
THEOCRACY
DEMOCRACY
TOTALITARIANISM
TECHNOCRACY
GESTALT MIND
GERONTOCRACY
GALACTIC FEDERATION
UNIVERSAL EMPIRE

**Comment by MattOttawa** [comment]
I never really been sure what to do with this. It might need more fixing up. Xml names should be checked. All I wanted to do was turn it on.
I kinda wanna get this pr out of the way now.
I'm thinking I'll commit this the way it is (turned on) and see if there are any comments from players.
If players (or DH) say it's not working good or don't like it then I'll disable it in python config folder.
Yudishtira said he played with this on and it works. I think... was a while ago.

**Comment by dzfischer** [comment]
Wasn't this in C2C previously? Somehow I remember playing with it, though might've been AND... was a while ago.

If it seems to work without glaring bugs, give it a shot- I'm just about to slow down modding a bit to play a game + check to see how my tech diffusion edits feel, so would be happy to play with this enabled to see how it feels/acts in gameplay (without revolutions, tho).

**Comment by Thunderbrd** [comment]
Yes it was in C2C but AIAndy was trying to improve it and did a lot of rewiring then never debugged his design.

**Comment by MattOttawa** [comment]
Ah yes, I think part of it was he was gonna add some xml for this.
I'm just doin a quick test. 
FAssert: info type 'CIVIC_HEREDITARY_RULE' not found
FAssert: info type 'CIVIC_FASCIST' not found
FAssert: info type 'CIVIC_FEDERAL' not found
Is there somthing similar I can change these to?

**Comment by raxo2222** [comment]
@MattOttawa you need to update that branch since its extremely outdated.
Civics (and other things) were renamed in March.

**Comment by MattOttawa** [comment]
CIVIC_HEREDITARY_RULE -> MONARCHY
CIVIC_FASCIST -> TOTALITARIANISM
I'm not sure what CIVIC_FEDERAL would be now.

**Comment by raxo2222** [comment]
That would be CIVIC_FEDERALISM now

**Pull request merged**